sdk/react-forms/src/lib/ProtectedForm.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { ComponentProps, ForwardedRef, forwardRef, useEffect, useMemo, useRef, useState } from 'react';
+import { createProtectedForm, type Form, type ProtectedFormOptions, type ResultType } from '@piiano/forms';
+
+export function useProtectedForm<T extends ResultType = 'fields'>(options: ProtectedFormOptions<T>) {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const [form, setForm] = useState<Form<T>>();
+
+  useEffect(() => {
+    if (!containerRef.current) return;
+
+    if (!form) {
+      const form = createProtectedForm(containerRef.current, options);
+      setForm(form);
+    } else {
+      form.update(options);
+    }
+  }, [options]);
+
+  return { containerRef, form };
+}
+
+type ProtectedFormProps<T extends ResultType> = Omit<ProtectedFormOptions<T>, 'hooks'> &
+  ProtectedFormOptions<T>['hooks'];
```

Comment:
Are you omitting hooks and then adding it back? what?

Reply:
The forms options looks like:
`{ vaultURL: '', apiKey: '', ..., hooks: { onSubmit: (value) => {}, onError: (err) => {} } }`
This flatten it to something like this:
`{ vaultURL: '', apiKey: '', ..., onSubmit: (value) => {}, onError: (err) => {} }`
The reason is in react it makes more sense to have these things as direct attributes of the component.

Reply:
ok. Is that a common practice? if not, can you put a comment for smooth brains like me? :)